RefactoringPracticeI/GameInventory/Inventory/GameInventory.cs

```diff
@@ -11,66 +11,21 @@ public GameInventory(IList<Item> Items)
 
         public void UpdateQuality()
         {
-            for (var index = 0; index < _items.Count; index++)
+            foreach (Item item in _items)
             {
-                if (_items[index].Name != "Aged Brie" && _items[index].Name != "Backstage passes to a Pokemon Gym concert")
-                {
-                    if (_items[index].Quality > 0 && _items[index].Name != "Sulfuras, Hand of Ragnaros")
-                    {
-                        _items[index].Quality--;
-                    }
-                }
-                else
-                {
-                    if (_items[index].Quality < 50)
-                    {
-                        _items[index].Quality++;
-
-                        if (_items[index].Name == "Backstage passes to a Pokemon Gym concert")
-                        {
-                            if (_items[index].SellIn < 11 && _items[index].Quality < 50)
-                            {
-                                _items[index].Quality++;
-                            }
-
-                            if (_items[index].SellIn < 6 && _items[index].Quality < 50)
-                            {
-                                _items[index].Quality++;
-                            }
-                        }
-                    }
-                }
+                Item result;
 
-                if (_items[index].Name != "Sulfuras, Hand of Ragnaros")
+                result = item.Name switch
                 {
-                    _items[index].SellIn--;
-                }
-
-                if (_items[index].SellIn < 0)
-                {
-                    if (_items[index].Name != "Aged Brie")
-                    {
-                        if (_items[index].Name != "Backstage passes to a Pokemon Gym concert")
-                        {
-                            if (_items[index].Quality > 0 && _items[index].Name != "Sulfuras, Hand of Ragnaros")
-                            {
-                                _items[index].Quality--;
-                            }
-                        }
-                        else
-                        {
-                            _items[index].Quality = _items[index].Quality - _items[index].Quality;
-                        }
-                    }
-                    else
-                    {
-                        if (_items[index].Quality < 50)
-                        {
-                            _items[index].Quality++;
-                        }
-                    }
-                }
+                    "Aged Brie" => new AgedBrie(item).Update(),
+                    "Backstage passes to a Pokemon Gym concert" => new BackstagePokemon(item).Update(),
+                    "Sulfuras, Hand of Ragnaros" => item,
+                    _ => new Default(item).Update(),
```

Comment:
Gotta like the dummy variable (_) just make sure that the language version wherever you work supports these nice utils (for instance C# ver. 7+ in this case). 

We had a problem a while back where we were using string interpolation, and it was not supported by the C# version we were using.